astroplan/constraints.py

```diff
@@ -467,7 +566,91 @@ def compute_constraint(self, times, observer, targets):
         else:
             mask = [(t.datetime.time() >= min_time) or
                     (t.datetime.time() <= max_time) for t in times]
+        if targets is not None:
+            mask = np.tile(mask, len(targets))
+            mask = mask.reshape(len(targets), len(times))
+        return np.atleast_2d(mask)
+
+
+class UtcDateConstraint(Constraint):
```

Comment:
I'm sure you have an idea in mind, but would you mind including an example in the docstring for what makes the `UtcDateConstraint` useful instead of setting your time limits in the `is_observable` or `is_always_observable` etc. function? Those functions take a time or time-range argument that's degenerate with this constraint. 

Were you thinking about using this constraint object for scheduling downstream, e.g. so that you can associate a specific time-range with a specific observing block? This would be a very useful class for that purpose. It would be great if the docstring made that distinction, by something something like: "this class is useful for associating an acceptable time-range with an observing block."

***

I would also advocate for calling this the `TimeConstraint`, because it doesn't require the times that you input to be UTC, they just need to be an astropy `Time` object.